src/enzyme_ad/jax/Passes/AffineToStableHLORaising.cpp

```diff
@@ -1523,21 +1610,41 @@ tryRaisingOpToStableHLO(Operation *op, IRMapping &mapping, OpBuilder &builder,
           affine::AffineValueMap(AffineMap::get(op->getContext()), {});
 
       for (auto &innerOp : forOp.getBody()->without_terminator()) {
-        if (tryRaisingOpToStableHLO(&innerOp, mapping, builder, maps).failed())
+        if (tryRaisingOpToStableHLO(&innerOp, mapping, builder, maps, pc)
+                .failed())
           return failure();
       }
 
       Value newIvInBody =
           builder.create<stablehlo::AddOp>(forOp.getLoc(), ivInBody, step);
 
       SmallVector<Value> loopCarried = {newIvInBody};
+      for (auto [iterArg, yieldedIterArgs] :
+           llvm::zip(forOp.getRegionIterArgs(),
+                     forOp.getBody()->getTerminator()->getOperands())) {
+        if (maps[mapping.lookup(iterArg)].getOperands() !=
+            maps[mapping.lookup(yieldedIterArgs)].getOperands()) {
```

Comment:
```suggestion
        if (maps[mapping.lookup(iterArg)] !=
            maps[mapping.lookup(yieldedIterArgs)]) {
```

Thank you very much for the fixes.

If the maps themselves must match then this should be better?